app/javascript/controllers/top_controls_controller.js

```diff
@@ -185,4 +200,86 @@ export default class extends Controller {
   #isMobile() {
     return window.innerWidth < 640 // Tailwind's sm breakpoint
   }
+
+  initializeTashkeelToggle() {
+    if (this.hasContentTarget) {
+      // Store the original content if we have tashkeel toggle buttons
+      if (this.hasTashkeelToggleButtonTarget) {
```

Comment:
No need for these checks, StimulusJS raises an error if there is a missing target.